backend/src/modules/room/room.controller.ts

```diff
@@ -0,0 +1,99 @@
+import { Body, Controller, Get, HttpStatus, Logger, Param, Post, Query, Res, UseGuards, ValidationPipe } from '@nestjs/common';
+import { RoomService } from './room.service';
+import { CreateRoomDTO } from './dto/createRoom';
+import { Response } from 'express';
+import { AuthorizeJWT } from '../../common/guards/authorizeJWT';
+import { ResponseClient } from '../../common/types/responseClient';
+import { extractIdRoom } from '../../common/utils/helper';
+import { RoomUserService } from '../roomUser/roomUser.service';
+import { IdUser } from '../../common/decorators/idUser';
+
+@Controller('room')
+export class RoomController {
+  constructor(
+    private roomService: RoomService,
+    private logger: Logger = new Logger(RoomController.name),
+    private roomUserService: RoomUserService,
+  ) { }
+
+  @UseGuards(AuthorizeJWT)
+  @Post('/create')
+  async createRoom(
+    @Body(new ValidationPipe()) roomInformation: CreateRoomDTO,
+    @Res() response: Response,
+    @IdUser() idUser: number,
+  ) {
+    try {
+      const newRoom = await this.roomService.createNewRoom({
+        ...roomInformation,
+        host: idUser,
+      });
+
+      return response.status(HttpStatus.CREATED).json({
+        message: 'Create new room successfully!',
+        statusCode: HttpStatus.CREATED,
+        success: true,
+        data: newRoom,
+      } as ResponseClient);
```

Comment:
[IMPORTANCE] - [REST API RESPONSE]
@TravisMai @hnvkhanh @trungnguyenhuynhminh46 @MiraiJR @khoatruong19 @lequocanhh

- Đừng nên bọc data lại bởi các từ như: `data`, `response`, `info`, ...
- Khi suscess thì cũng không cần `message` hay `status code` trong JSON response nữa. Nó sẽ bị thừa

Tụi em có thể làm theo như sau:
1. `POST /rooms`
```json
{
  "id": 1,
  "name": "Room A",
  "title": "Conference Room",
  "code": "CONF1"
}
```

2. `GET /rooms/:id`
```json
{
  "id": 1,
  "name": "Room A",
  "title": "Conference Room",
  "code": "CONF1"
}
```

3. `PUT /rooms/:id`
```json
{
  "id": 1,
  "name": "Room B",
  "title": "Meeting Room",
  "code": "MEET1"
}
```

4. `DELETE /rooms/:id`
204 status, empty body

5. `GET /rooms`
```json
[
  {
    "id": 1,
    "name": "Room A",
    "title": "Conference Room",
    "code": "CONF1"
  },
  {
    "id": 2,
    "name": "Room B",
    "title": "Meeting Room",
    "code": "MEET1"
  }
]
```

Trong trường hợp API trả thêm nhiều fields ví dụ meta data (link to next page, link to previous page), phân trang, tổng số trang, ... thì tụi em mới phải bọc nó lại. Có thể bọc lại với chữ `rooms`, hoặc thật ra vẫn có nơi bọc lại bằng `data` nhưng anh không thích dùng mấy từ chung chung :))) Cái này cũng tuỳ vào convention của mỗi project

Ví dụ:
`GET /rooms?page=2&size=2`
```json
{
  "rooms": [
    {
      "id": 4,
      "name": "Room D",
      "title": "Conference Room",
      "code": "CONF4"
    },
    {
      "id": 5,
      "name": "Room E",
      "title": "Meeting Room",
      "code": "MEET5"
    },
  ],
  "page": 2,
  "size": 2,
  "totalPages": 10,
  "totalItems": 30,
}
```